File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {

        Set<Integer> set = new HashSet<>();

        for  (int n: nums) {
            if (set.contains(n)) {
                return true;
            } else {
                set.add(n);
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set(nums.begin(), nums.end());

        int longest = 0;

        for( auto num : nums) {
            if(!set.contains(num - 1)) {
                int curr_num = num;
                int curr_length = 1;

                while(set.contains(curr_num + 1)) {
                    curr_num ++;
                    curr_length ++;
                }

                longest = max(longest, curr_length);
            }
        }

        return longest; 
    }
};
